src/Orleans.Core/Configuration/Options/MessagingOptions.cs

```diff
@@ -11,8 +11,20 @@ public abstract class MessagingOptions
         /// <summary>
         /// The ResponseTimeout attribute specifies the default timeout before a request is assumed to have failed.
         /// </summary>
-        public TimeSpan ResponseTimeout { get; set; } = DEFAULT_RESPONSE_TIMEOUT;
-        public static readonly TimeSpan DEFAULT_RESPONSE_TIMEOUT = Debugger.IsAttached ? TimeSpan.FromMinutes(30) : TimeSpan.FromSeconds(30);
+        public TimeSpan ResponseTimeout
+        {
+            get { return (DebuggerOverridesTimeout && Debugger.IsAttached) ? RESPONSE_TIMEOUT_WITH_DEBUGGER : responseTimeout; }
+            set { this.responseTimeout = value; }
+        }
+        public static readonly TimeSpan DEFAULT_RESPONSE_TIMEOUT = TimeSpan.FromSeconds(30);
+        private TimeSpan responseTimeout = DEFAULT_RESPONSE_TIMEOUT;
+
+        /// <summary>
+        /// By default, the value from<see cref="ResponseTimeout"/> will be ignored and set to 30 minutes
+        /// if the debugger is attached. Set this flag to false to always use <see cref="ResponseTimeout"/>
+        /// </summary>
+        public bool DebuggerOverridesTimeout { get; set; } = true;
+        public static readonly TimeSpan RESPONSE_TIMEOUT_WITH_DEBUGGER = TimeSpan.FromMinutes(30);
```

Comment:
It seems that we should just lift the whole debug override code out to the top level so we can set it in our tests and others in theirs